sftplib/src/main/java/com/freiheit/fuava/sftp/SftpOldFilesMovingLatestMultiFileFetcher.java

```diff
@@ -79,6 +81,29 @@ public SftpOldFilesMovingLatestMultiFileFetcher(
         this.skippedFolder = skippedFolder;
         this.processingFolder = processingFolder;
         this.incomingFilesFolder = incomingFilesFolder;
+        this.moveToProcessing = true;
```

Comment:
I would allow null for the processing folder and calculate `moveToProcessing` based on this. Then only call `this( remoteClient, skippedFolder, null, incomingFilesFolder )` in the newly created constructor.

Additionally I would propose to also add some `Preconditions.checkNotNull`s here.